libs/ml/include/ml/ops/weights.hpp

```diff
@@ -107,6 +140,27 @@ class Weights : public fetch::ml::ops::PlaceHolder<T>, public Trainable<T>
     ResetGradients();
   }
 
+  virtual void L1Regularisation(DataType regularisation_rate)
+  {
+    ArrayType weight     = *this->output_;
+    ArrayType reg_output = ArrayType(weight.shape());
+
+    fetch::math::Multiply(weight, static_cast<DataType>(-2) * regularisation_rate, reg_output);
```

Comment:
static casting -2 to a DataType is dangerous if -2 is ever an unsigned integer. I think we should probably template guard graph from permitting that use case though